cursive-core/src/printer.rs

```diff
@@ -118,16 +115,53 @@ impl<'a, 'b> Printer<'a, 'b> {
         }
     }
 
+    /// Prints some text at the given position.
+    ///
+    /// # Parameters
+    /// `start` is an offset used to print the text in the view.\
+    /// `text` is a simple string to print on the screen.
+    ///
+    /// # Description
+    /// Prints some text at the given position.
+    /// The text could be truncated if it exceed the [drawing area size](Self::output_size).
+    ///
+    /// # Example
+    /// ```rust
+    /// use cursive::{View, Printer, XY, Vec2};
+    /// 
+    /// pub struct CustomView {
+    ///     word: String,
+    /// }
+    /// 
+    /// impl CustomView {
+    ///     pub fn new() -> Self {
+    ///         Self {
+    ///             word: String::from("Eh, tu connais Rust?"),
+    ///         }
+    ///     }
+    /// }
+    /// 
+    /// impl View for CustomView {
+    ///     fn draw(&self, printer: &Printer<'_, '_>){
+    ///         printer.print(XY::new(0, 0), self.word.as_str());
```

Comment:
```suggestion
    ///         printer.print(XY::new(0, 0), &self.word);
```

I think the example could be reduced to only include a `draw` function, to focus on the `print` method (and not on implementing a view).

Reply:
If I want to implement the `required_size` method here, it's because if you are new with your library, then you could be in trouble to understand why your call to the `print` method leads to a bit of an "unexpected behavior" (by cutting the `text` before its end).

Reply:
I agree, but this is a concern of implementing your own view (which could be part of the View documentation). I don't think we should mention this in _every_ example for the printer.

As for `&` vs `.as_str`, it's _quite_ idiomatic to use the `Deref<Target=str>` from `String`, I don't think anyone would be surprised.

Reply:
Ok, so I will remove the `as_str` call. Maybe I can warn in the global Printer class documentation, about the behavior of the printer on the views?

Reply:
This is really something that belongs in the view - you only care about that when you implement your own view.

We _can_ mention in the doc that anything further than the printer's `size` would be cropped out.

Reply:
Ok, I've done the modification on my last commits.

It could be good to mention that in the `Printer` global documentation too, maybe (`Printer` not strictly related to `View`). And maybe that's why Printer's members stay a bit... obscure to me.

Reply:
As I understand what you are saying, `Printer` is always related to one view, but a single view can "contains" one or more printers.
Do you have an example of a `Printer` that works without any view?

Reply:
@gyscos Am I right ?
Do you ant me to add this to the documentation, or do you want to do it yourself ?

Reply:
In practice `Printer` is always "associated" with a view, though this association is merely implicit (it just happens to be given to the view in its `draw` function).

The _concept_ of Printer doesn't _need_ views. For example for some time the menubar was not really a `View`, but was still using the `Printer`. Same for the background. Today these have been made into views, but that's more a detail.

I don't think it's very important though. We may add doc about it later if it becomes useful.